### PR TITLE
modify func signatures refer PCP #13973

### PR DESCRIPTION
**Pull request by ZhaoQian888**
https://github.com/pingcap/tidb/pull/14128

Separate signatures for PlusInt

### CODE REVIEW

File: proto/expression.proto [hunk 1]
```diff
@@ -207,7 +207,6 @@ enum ScalarFuncSig {
 	/*arithmetic*/
 	PlusReal = 200;
 	PlusDecimal = 201;
-	PlusInt = 203;
```
Comment (breeswish) [inline proto/expression.proto]: Hi, please reserve the original signature so that TiKV can be compatible with old TiDB.

Comment (ZhaoQian888) [reply proto/expression.proto]: ok

File: proto/expression.proto [hunk 2]
```diff
@@ -222,6 +222,11 @@ enum ScalarFuncSig {
 	ModDecimal = 216;
 	ModInt = 217;
 	MultiplyIntUnsigned = 218;
+	PlusIntUnsignedUnsigned  = 219;
+	PlusIntUnsignedSigned   = 220;
+	PlusIntSignedUnsigned   = 221;
+	PlusIntSignedSigned     = 222;
```
Comment (breeswish) [inline proto/expression.proto]: Prefer to be `PlusIntSignedSigned = 222;` instead to be aligned with styles above.

**Comment by breeswish** [comment]
@winoros @lonng  PTAL

**Comment by breeswish** [comment]
@lonng @qw4990 PTAL, thanks!
@ZhaoQian888 Please resolve conflicts, thanks!

**Comment by ZhaoQian888** [comment]
ok, I think conflict has been solved

**Pull request merged**